pulser-core/pulser/sequence/sequence.py

```diff
@@ -1802,6 +1817,14 @@ def _validate_channel(
             raise ValueError("Use the name of a declared channel.")
         if block_eom_mode and self.is_in_eom_mode(channel):
             raise RuntimeError("The chosen channel is in EOM mode.")
+        if (
+            channel == self._slm_mask_dmm
+            and self._schedule[self._slm_mask_dmm].get_duration() == 0
```

Comment:
Two issues:
- `_validate_channel` is more of a way to check if a channel is valid, I think going further to block adding things to it is assuming it is only being called before a change. I would be more comfortable if that were controlled by an argument (like `block_eom_mode`) so we don't modify the default behaviour.
- Calling `_ChannelSchedule.get_duration()` repeatedly might be a bit time consuming, maybe it'd make sense to have a flag registering whether the first pulse has been added instead? (Only if it is clear this flag will always be reliably switched, otherwise it's not worth it)

Reply:
Thanks for these observations, errors could be induced from this indeed.